hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/impl/prefetch/SingleFilePerBlockCache.java

```diff
@@ -67,6 +71,8 @@ public class SingleFilePerBlockCache implements BlockCache {
 
   private final PrefetchingStatistics prefetchingStatistics;
 
+  private static LocalDirAllocator localDirAllocator;
```

Comment:
shouldn't be static, as that would be shared across all fs instances, even those of different users. 

better: constructor takes an allocator, which is created in s3a caching block manager.